### PR TITLE
Add installation instructions on Alpine Linux and FreeBSD

### PR DESCRIPTION
**Pull request by straight-shoota**
/cc @jirutka @myfreeweb please take a look, I might've missed something.

### CODE REVIEW

File: installation/on_alpine_linux.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+Alpine Linux includes the Crystal compiler in the community repository, starting from version Alpine 3.7.
+
+Currently, the package is only available for `aarch64` and `x86_64` platforms.
+
+## Install
+
+```bash
+sudo apk update
+sudo apk add crystal@community shards@community
```
Comment (RX14) [inline installation/on_alpine_linux.md]: I'm pretty sure you don't need the `@community`

Comment (straight-shoota) [reply installation/on_alpine_linux.md]: Me too, but it failed to work without it.

Comment (RX14) [reply installation/on_alpine_linux.md]: It works fine for me

Comment (straight-shoota) [reply installation/on_alpine_linux.md]: Oh, I figured it out. The vagrant box I used (`generic/alpine38`) has `@community` pinned in `/etc/apk/repositories`. That's why it was needed.

**Comment by j8r** [comment]
You have used the `bash` tag instead of `sh`  for shell commands to be executed on non-GNU Unixes @straight-shoota, curse!

**Comment by straight-shoota** [comment]
@j8r Yes, that'e because from my experience `sh` doesn't always work as expected, whereas `bash` does. And it's really just the highlighter...

**Comment by j8r** [comment]
That's because `sh` may refer to various shells, including `bash`.
Indeed `sh` and `bash` are the same highlighter on GitHub 😄

File: installation/on_alpine_linux.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# On Alpine Linux
+
+Alpine Linux includes the Crystal compiler in the community repository, starting from version Alpine 3.7.
+
+Currently, the package is only available for `aarch64` and `x86_64` platforms.
+
+## Install
+
+```bash
+sudo apk update
```
Comment (jirutka) [inline installation/on_alpine_linux.md]: This is unnecessary, apk automatically updates indexes if they're stale.

File: installation/on_alpine_linux.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+sudo apk add crystal shards
```
Comment (jirutka) [inline installation/on_alpine_linux.md]: I'd omit sudo. It's not required to run apk via sudo. It must be run as root, but there are many ways how to achieve it and sudo is not even part of the Alpine base.

**Comment by straight-shoota** [comment]
Thanks @jirutka 👍

File: installation/on_freebsd.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+FreeBSD includes the Crystal compiler in the ports tree, starting from version FreeBSD 11.0.
+
+Currently, it is only available for `aarch64` and `amd64` platforms.
+
+When building Crystal code with the `--release` flag on FreeBSD, the `--no-debug` flag should be added too in order to avoid LLVM assertion errors.
+
+## Install Package
+
+Crystal is available as a compiled package. However, it might not be the most recent version available.
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply installation/on_freebsd.md]: Well, currently it seems the version in the ports tree is more recent than the binary.

[the remark this answers is not in the document]
Comment (straight-shoota) [reply installation/on_freebsd.md]: It could be rephrased to be less "scary".

**Comment by RX14** [comment]
A squash would be nice

**Comment by RX14** [comment]
@straight-shoota also can you please stop committing when time-travelling, thanks :)

**Comment by straight-shoota** [comment]
I'll try 😆 Dunno why git used a time stamp one hour in the future... system clock seems to be fine. It doesn't reproduce either.

**Comment by bcardiff** [comment]
Is there a way to install crystal from sources, in order to get the latest version but use binaries packages for llvm and other dependencies? 

It takes ages the "Install Port" instructions. 😅

**Comment by straight-shoota** [comment]
I'm not too confident with the alpine build process, but I guess you can checkout https://github.com/alpinelinux/aports and change the source version in `community/crystal/APKBUILD`. Then it's just running `abuild checksum && abuild -r` to build from source.

**Comment by myfreeweb** [comment]
@bcardiff oh — you just need to install LLVM itself from binary packages (`pkg install llvm60`) and *then* install Crystal from ports.

**Pull request merged**